portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -66,16 +76,33 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     Entity messageEntity = new Entity("Message");
     messageEntity.setProperty(TIME, timestamp);
     messageEntity.setProperty(NEWEST_MESSAGE, fullMessage);
+
     datastore.put(messageEntity);
 
     response.sendRedirect("/write-message.html");
   }
 
   private String getParameter(HttpServletRequest request, String name, String defaultValue) {
     String value = request.getParameter(name);
-    if (value == null) {
+    if (value == null || value.trim().isEmpty()) {
       return defaultValue;
     }
     return value;
   }
+
+  private int convertToInt(String beingconverted) {
+    int convertee;
+    try {
+      convertee = Integer.parseInt(beingconverted);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + beingconverted);
+      return 5;
```

Comment:
Instead of returning 5 on error, returning null would be more clear (how would callers differ between valid and invalid input?). Then have the caller handle the null case. Also, please add javadoc comment to the method about the valid range.